### DISCUSSION TITLE
NEXTAUTH_URL is set, but still redirects to "commit-url"

### DISCUSSION THREAD

**Discussion by Glutch** [nextauthjs/next-auth#1317] [Help]
I have published a simple nextauth-user-test website to vercel with discord as only provider.

**What i do and what happens**
1. Not signed in [Sign in]
2. Redirected to pre-built signin page
3. Click on "sign in with discord"
4. Gets redirected to the SAME PAGE but on my vercel latest commit url "https://project-pq69ayfdp.vercel.app/"
5. Click on "sign in with discord" AGAIN
6. The return_uri is now WRONG because it has pq69ayfdp in it
7. Discord says "invalid Oath2 redirect_uri"

How do i solve this?

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [ ] Found the documentation helpful
* [x] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [x] Did not find the example project helpful

**Comment by StefanSelfTaught** [comment]
Did you set your `NEXTAUTH_URL` environment variable on vercel? And maybe you want your `redirect_uri` to be the same with `NEXTAUTH_URL`
